src/IO/S3Common.cpp

```diff
@@ -599,7 +599,13 @@ class S3CredentialsProviderChain : public Aws::Auth::AWSCredentialsProviderChain
     {
         auto * logger = &Poco::Logger::get("S3CredentialsProviderChain");
 
-        if (use_environment_credentials)
+        /// add explicit credentials to the front of the chain
+        /// because it's manually defined by the user
+        if (!credentials.IsEmpty())
+        {
+            AddProvider(std::make_shared<Aws::Auth::SimpleAWSCredentialsProvider>(credentials));
```

Comment:
I think after `AddProvider()` here we can just return because it doesn't look we need to add `ProfileConfigFileAWSCredentialsProvider` if we have explicit credentials.

Reply:
I tried to leave as same as possible the old behavior but actually if we have explicit behavior we will never go to a different provider down the chain because we know the first one is not empty so yeah it can be return.